### ISSUE TITLE
Add support for Micron cards

### ISSUE THREAD

**Issue by J-Twin** [Ognian/sdmon#15]
Hello,

Micron provides public documentation regarding acquisition of health condition for its cards here:
https://www.micron.com/-/media/client/global/documents/products/technical-note/sd-cards/tnsd02_enable_sd_health_monitor_system.pdf

To support those devices, I copied Longsys,Sandisk code, modified it and added below them the following: 

//try micron argument
  cmd56_arg = 0x110005fb;
  ret = CMD56_data_in(fd, cmd56_arg, data_in);
  // we assume success when the call was successful AND the signature is not 0xff 0xff
  if (ret == 0 && !((data_in[0] == 0xff && data_in[1] == 0xff) || (data_in[0] == 0x00 && data_in[1] == 0x00))) {
    printf("\"signature\":\"0x%x 0x%x\",\n", data_in[0], data_in[1]);

    if (data_in[0] == 0x4d && data_in[1] == 0x45) {
      printf("\"Micron\":\"true\",\n");
      printf("\"Percentange step utilization\": %d,\n", (int)(data_in[7]));
      printf("\"TLC area utilization\": %d,\n", (int)(data_in[8]));
      printf("\"SLC area utilization\": %d,\n", (int)(data_in[9]));
      close(fd);
      printf("\"success\":true\n}\n");
      exit(0);
   }
  }

Compiled it and tested it on a brand new i400 card on my T6 ARM board:

sudo ./sdmonm /dev/mmcblk0
{
"version": "v0.6.0 (dba7898) arm64",
"date": "2023-12-07T22:00:21.000Z",
"device":"/dev/mmcblk0",
"addTime": "false",
"signature":"0x4d 0x45",
"Micron":"true",
"Percentange step utilization": 1,
"TLC area utilization": 0,
"SLC area utilization": 0,
"success":true
}

**Comment by Ognian** [comment]
Thanks a lot!
I merged this in the latest build. Would you test, since I have no micron cards?
Ognian

**Comment by J-Twin** [comment]
Hello, just tested it.

It is not 100% correct but we are close!

Here is what I got:
sudo ./sdmon /dev/mmcblk0
{
"version": "v0.6.0-7 (be9af24) arm64",
"date": "2023-12-10T21:14:37.000Z",
"device":"/dev/mmcblk0",
"addTime": "false",
"read_via_cmd56_arg_1":"read successful but signature 0xff 0xff",
"signature":"0x 4d 0x 45",
"Micron":"true",
"Percentange step utilization": 1,
"TLC area utilization": 0,
"SLC area utilization": 0,
"success":true
}

While the code I tested was added just after this:
      close(fd);
      printf("\"success\":true\n}\n");
      exit(0);
    }

I can see it after this in your repository:
  if (ret == 0) {
    printf("\"read_via_cmd56_arg_1\":\"read successful but signature 0x%x 0x%x\",\n", data_in[0], data_in[1]);
  } else {
    printf("\"read_via_cmd56_arg_1\":\"not implemented: %s\",\n", strerror(errno));
  }

If you move it up 6 lines it will be fine. Signature is not ff for sure as you can see it get printed shortly afterwards.

Once you change it I will test it again!

**Comment by Ognian** [comment]
Should be fixed now...
would you test again?

**Comment by J-Twin** [comment]
Sure no problem.

It is OK now:
sudo ./sdmon /dev/mmcblk0
{
"version": "v0.6.0-21 (04e39e7) arm64",
"date": "2023-12-11T17:52:51.000Z",
"device":"/dev/mmcblk0",
"addTime": "false",
"signature":"0x4d 0x45",
"Micron":"true",
"Percentange step utilization": 1,
"TLC area utilization": 0,
"SLC area utilization": 0,
"success":true
}

**Comment by Ognian** [comment]
released with v0.7.0

**Comment by Fliegefisch** [comment]
> Should be fixed now... would you test again?

Thank you both for implementing. I never got cmd56 to work when I tried it about three years ago. Today I found your software.


{
"version": "v0.6.0-21 (04e39e7) armv7",
"date": "2023-12-11T16:15:25.000Z",
"device":"/dev/mmcblk0",
"addTime": "false",
" signature ":"0x4d 0x45",
"Micron":"true",
"Percentange step utilization": 1,
"TLC area utilization": 2,
"SLC area utilization": 0,
"success":true
}
